src/Atmos/Model/source.jl

```diff
@@ -173,3 +181,69 @@ function atmos_source!(
         source.ρu -= β_sponge * (state.ρu .- state.ρ * s.u_relaxation)
     end
 end
+
+"""
+    RemovePrecipitation{FT} <: Source
+A sink to `q_tot` when cloud condensate is exceeding a threshold.
+The threshold is defined either in terms of condensate or supersaturation.
+The removal rate is implemented as a relaxation term
+in the Microphysics_0M module.
+The default thresholds and timescale are defined in CLIMAParameters.jl.
+"""
+struct RemovePrecipitation <: Source
+    " Set to true if using qc based threshold"
+    use_qc_thr::Bool
+end
+function atmos_source!(
+    s::RemovePrecipitation,
+    atmos::AtmosModel,
+    source::Vars,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    direction,
+)
+    # TODO - how to assert that it is used with EquilMoist module?
+    #        (if it were to be used with non-EquilMoist model
+    #         q_liq and q_ice would be part of state)
```

Comment:
```suggestion
atmos_source!(
    s::RemovePrecipitation,
    atmos::AtmosModel,
    source::Vars,
    state::Vars,
    diffusive::Vars,
    aux::Vars,
    t::Real,
    direction,
) = atmos_source!(s, atmos, atmos.moisture, source, state, diffusive, aux, t, direction)

function atmos_source!(
    s::RemovePrecipitation,
    atmos::AtmosModel,
    moist::EquilMoist,
    source::Vars,
    state::Vars,
    diffusive::Vars,
    aux::Vars,
    t::Real,
    direction,
)
```

Reply:
Probably need to format this, but this is how we can restrict to the `EquilMoist` model

Reply:
Thanks! Now that I'm not using `aux.q_liq` and `aux.q_ice`. I think this is not needed. `thermo_state` should do the job of switching between the Equil and NonEquil moisture models.